File: wallendar-app/app/src/main/java/com/td/wallendar/home/profile/ui/ProfileViewImpl.java
package com.td.wallendar.home.profile.ui;

import static android.view.Gravity.CENTER;

import android.content.Context;
import android.util.AttributeSet;
import android.view.View;
import android.widget.LinearLayout;
import android.widget.TextView;

import androidx.annotation.Nullable;
import androidx.appcompat.app.AlertDialog;

import com.td.wallendar.R;
import com.td.wallendar.home.profile.OnLogoutClickedListener;
import com.td.wallendar.home.profile.OnShowAliasesClickedListener;
import com.td.wallendar.models.ApplicationUser;

public class ProfileViewImpl extends LinearLayout implements ProfileView {

    private OnShowAliasesClickedListener onShowAliasesClickedListener;
    private OnLogoutClickedListener onLogoutClickedListener;

    public ProfileViewImpl(Context context) {
        this(context, null);
    }

    public ProfileViewImpl(Context context, @Nullable AttributeSet attrs) {
        this(context, attrs, 0);
    }

    public ProfileViewImpl(Context context, @Nullable AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);

        inflate(context, R.layout.view_profile, this);
        setGravity(CENTER);
        setOrientation(VERTICAL);

    }

    @Override
    public void bind(ApplicationUser applicationUser,
                     OnShowAliasesClickedListener onShowAliasesClickedListener,
                     OnLogoutClickedListener onLogoutClickedListener) {
        this.onShowAliasesClickedListener = onShowAliasesClickedListener;
        this.onLogoutClickedListener = onLogoutClickedListener;

        TextView profileName = findViewById(R.id.profile_name);
        TextView profileEmail = findViewById(R.id.profile_email);
        String fullName = applicationUser.getFirstName() + " " + applicationUser.getLastName();

        profileName.setText(fullName);
        profileEmail.setText(applicationUser.getEmail());

        LinearLayout logoutButton = findViewById(R.id.layout_button_logout);
        logoutButton.setOnClickListener(v -> {
            onLogoutClickedListener.onLogoutClicked();
        });
    }

}

File: wallendar-app/app/src/main/java/com/td/wallendar/home/profile/ui/ProfileViewImpl.kt
package com.td.wallendar.home.profile.ui

import android.content.Context
import android.util.AttributeSet
import android.view.Gravity
import android.widget.LinearLayout
import android.widget.TextView
import com.td.wallendar.R
import com.td.wallendar.home.profile.OnLogoutClickedListener
import com.td.wallendar.home.profile.OnShowAliasesClickedListener
import com.td.wallendar.models.ApplicationUser

class ProfileViewImpl @JvmOverloads constructor(context: Context, attrs: AttributeSet? = null, defStyleAttr: Int = 0) : LinearLayout(context, attrs, defStyleAttr), ProfileView {
    private var onShowAliasesClickedListener: OnShowAliasesClickedListener? = null
    private var onLogoutClickedListener: OnLogoutClickedListener? = null
    override fun bind(
            applicationUser: ApplicationUser,
            onShowAliasesClickedListener: OnShowAliasesClickedListener,
            onLogoutClickedListener: OnLogoutClickedListener
    ) {
        this.onShowAliasesClickedListener = onShowAliasesClickedListener
        this.onLogoutClickedListener = onLogoutClickedListener
        val profileName = findViewById<TextView?>(R.id.profile_name)
        val profileEmail = findViewById<TextView?>(R.id.profile_email)
        val fullName = applicationUser.firstName + " " + applicationUser.lastName
        profileName.text = fullName
        profileEmail.text = applicationUser.email
        val logoutButton = findViewById<LinearLayout?>(R.id.layout_button_logout)
        logoutButton.setOnClickListener { onLogoutClickedListener.onLogoutClicked() }
    }

    init {
        inflate(context, R.layout.view_profile, this)
        gravity = Gravity.CENTER
        orientation = VERTICAL
    }
}